### ISSUE TITLE
Api documentation

### ISSUE THREAD

**Issue by frechina** [Payum/PayumServer#46]
Hi,

I've been wondering if there is some documentation about how to use the PayumServer API. I've been playing with the code and after solving some issues I'm stuck adding gateways. Any help? I wanted to add stripe and paypal.

Thank you

**Comment by makasim** [comment]
There is a js client for the server (which uses it api). https://github.com/Payum/PayumServerUI

You can start with it and see what requests it sends and what format.

The client is easy to do (especially if you gonna use docker image `payum/server-ui`. Once up and running go to settings and configure the server url to work with.
You can configure gateways there, purchase stuff, see payment details and so on. 

Once you have gateways configured you can open this small demo page.
https://github.com/Payum/PayumServer/blob/master/web/demo.html

It shows how the payment server could be integrated to your shop or app. 

Pay attention it is a proof of concept, there might be bugs (tell me if you find some) or missing features.

**Comment by makasim** [comment]
that's how I run it locally 

```
version: '2'
services:
  payum-server:
    image: payum/payum-server
    container_name: payum-server
    restart: always
    environment:
      - PAYUM_MONGO_URI=mongodb://mongo:27017/payum_server
      - PAYUM_DEBUG=1
      - VIRTUAL_HOST=payum-server.local
    links:
      - mongo
    volumes:
      - .:/payum

  payum-server-ui:
    image: payum/server-ui
    container_name: payum-server-ui
    restart: always
    environment:
      - VIRTUAL_HOST=payum-server-ui.local
    volumes:
      - ./PayumServerUI:/payum

  mongo:
    image: mongo
    restart: always
    volumes:
      - ./data:/data/db

  nginx-proxy:
    image: jwilder/nginx-proxy
    restart: always
    container_name: nginx-proxy
    ports:
      - "80:80"
    volumes:
      - /var/run/docker.sock:/tmp/docker.sock:ro
```

**Comment by frechina** [comment]
I tried to get the UI work without success. Also I don't understand that Authorize and Capture mean.
The demo.html doesn't work either :\

**Comment by frechina** [comment]
Moreover, this urls don't work

Demo: http://server.payum.org/demo.html
Backend: http://server.payum.org/client/index.html
Server: http://server.payum.org/

**Comment by makasim** [comment]
> Moreover, this urls don't work

I am in the process of migration them to another server. It will be available soon.

**Comment by makasim** [comment]
>I tried to get the UI work without success

Are there any errors? 
Did you use docker or setup it locally?

> Also I don't understand that Authorize and Capture mean.

It is not something invented by Payum but common terminology. Authorize means that the payment is authorized but any money is transferred anywhere. They could be kind of freezed on your account till you receive the goods. Capture means immediate  money transfer from one account to another.

**Comment by makasim** [comment]
> The demo.html doesn't work either :\

Did you see the page? 
Are there any errors? Maybe javascript errors in console log or os?

**Comment by frechina** [comment]
> I am in the process of migration them to another server. It will be available soon.

great! Thank you

> Are there any errors?
> Did you use docker or setup it locally?

I tried the docker and it needed some library installations, so the it wasn't fully configured.
There was js errors. For example, after configure the API url, it wasn't passed as a param and payments sections didn't work. After I realised this, I could put manually the param in the url and it worked. However, I couldn't complete a payment. I configured a paypal gateway and I got errors,

> Are there any errors? Maybe javascript errors in console log or os?
There was js errors. I finally could edit it and make it work
